src/nvim/eval.c

```diff
@@ -6860,8 +6860,10 @@ void screenchar_adjust_grid(ScreenGrid **grid, int *row, int *col)
       && *row < (msg_grid.rows + msg_grid.comp_row)
       && *col < msg_grid.cols) {
     *grid = &msg_grid;
-    *row -= msg_grid.comp_row;
```

Comment:
I think we can remove this special case at this point. ui_comp_get_grid_at_coord() should also handle msg grid

Reply:
Do you mean whole `if...` resulting in `*grid = &msg_grid;`?

Reply:
Done. But `msg_scroll_flush()`  is still needed (`oldtest` breaks without it). Also renamed helper function to `screenchar_adjust_coords()` which seems to be better suited.